File: android/dynamics-migration-tool/templates/sql/room-bridge/GDRoomStatement.java
// [BB_DYNAMICS-MIGRATION] This file was added as part of BlackBerry Dynamics migration.
// Provenance: kit-authored. License: Apache-2.0 — see templates/_LICENSE-NOTICE.md.

package __APP_PACKAGE__;

import androidx.annotation.NonNull;
import androidx.sqlite.db.SupportSQLiteStatement;

// [BB_DYNAMICS-MIGRATION] Dynamics SQLiteStatement — wraps compiled SQL inside the secure container.
import com.good.gd.database.sqlite.SQLiteStatement;

import java.io.IOException;

/**
 * SupportSQLiteStatement adapter wrapping a Dynamics SQLiteStatement.
 *
 * IMPORTANT — close() must declare throws IOException (from java.io.Closeable).
 */
// [BB_DYNAMICS-MIGRATION] Implements Room's SupportSQLiteStatement wrapping Dynamics SQLiteStatement.
final class GDRoomStatement implements SupportSQLiteStatement {

    private final SQLiteStatement stmt;

    GDRoomStatement(@NonNull SQLiteStatement stmt) {
        this.stmt = stmt;
    }

    // ---- Execution ----------------------------------------------------------

    @Override
    public void execute() {
        stmt.execute();
    }

    @Override
    public long executeInsert() {
        return stmt.executeInsert();
    }

    @Override
    public int executeUpdateDelete() {
        return stmt.executeUpdateDelete();
    }

    @Override
    public long simpleQueryForLong() {
        return stmt.simpleQueryForLong();
    }

    @Override
    public String simpleQueryForString() {
        return stmt.simpleQueryForString();
    }

    // ---- Bindings (from SupportSQLiteProgram) --------------------------------

    @Override
    public void bindNull(int index) {
        stmt.bindNull(index);
    }

    @Override
    public void bindLong(int index, long value) {
        stmt.bindLong(index, value);
    }

    @Override
    public void bindDouble(int index, double value) {
        stmt.bindDouble(index, value);
    }

    @Override
    public void bindString(int index, @NonNull String value) {
        stmt.bindString(index, value);
    }

    @Override
    public void bindBlob(int index, @NonNull byte[] value) {
        stmt.bindBlob(index, value);
    }

    @Override
    public void clearBindings() {
        stmt.clearBindings();
    }

    // IMPORTANT: throws IOException, NOT throws Exception.
    @Override
    public void close() throws IOException {
        stmt.close();
    }
}

File: android/dynamics-migration-tool/templates/sql/room-bridge/GDRoomStatement.kt
// [BB_DYNAMICS-MIGRATION] This file was added as part of BlackBerry Dynamics migration.
// Provenance: kit-authored. License: Apache-2.0 — see templates/_LICENSE-NOTICE.md.

package __APP_PACKAGE__

import androidx.sqlite.db.SupportSQLiteStatement
// [BB_DYNAMICS-MIGRATION] Dynamics SQLiteStatement — compiled SQL inside the secure container.
import com.good.gd.database.sqlite.SQLiteStatement
import java.io.IOException

// [BB_DYNAMICS-MIGRATION] Implements Room's SupportSQLiteStatement wrapping Dynamics SQLiteStatement.
internal class GDRoomStatement(private val stmt: SQLiteStatement) : SupportSQLiteStatement {

    override fun execute() = stmt.execute()
    override fun executeInsert(): Long = stmt.executeInsert()
    override fun executeUpdateDelete(): Int = stmt.executeUpdateDelete()
    override fun simpleQueryForLong(): Long = stmt.simpleQueryForLong()
    override fun simpleQueryForString(): String? = stmt.simpleQueryForString()

    override fun bindNull(index: Int) = stmt.bindNull(index)
    override fun bindLong(index: Int, value: Long) = stmt.bindLong(index, value)
    override fun bindDouble(index: Int, value: Double) = stmt.bindDouble(index, value)
    override fun bindString(index: Int, value: String) = stmt.bindString(index, value)
    override fun bindBlob(index: Int, value: ByteArray) = stmt.bindBlob(index, value)
    override fun clearBindings() = stmt.clearBindings()

    // IMPORTANT: throws IOException (from Closeable), NOT throws Exception.
    @Throws(IOException::class)
    override fun close() = stmt.close()
}
